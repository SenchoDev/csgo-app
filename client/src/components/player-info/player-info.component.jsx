import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import Settings from "../settings-overview/settings-overview.component";
import InfoButton from "../info-buton/info-button.component";
import Copy from '../Copy/copy.component'

import { selectPlayersInfo } from "../../redux/teams/teams.selector";
import { togglePlayersInfoHidden } from "../../redux/teams/teams.actions";
import { addCrosshair } from "../../redux/settings/settings.actions";
import { selectIfCopied } from "../../redux/settings/settings.selector";

import "./player-info.styles.scss";

import Target from "../../assets/target.png";
import Computer from "../../assets/computer.png";
import RedHeart from "../../assets/pink.png";
import PurpleHeart from "../../assets/purple.png";
import Close from "../../assets/close.png";

const PlayerInfo = ({
  info: { name, img, settings, role, crosshair, team, id },
  togglePlayersInfoHidden,
  copied,
  addCrosshair
}) => {
  return (
    <div className="info">
    <div className={`${copied ? 'info__copied-succes': ''} info__copied`}> <p className="info__copied-text">Copied to clipboard</p></div>
      <div className="info__header">
        <h3 className="info__heading">{name}</h3>
        <img
          src={Close}
          alt="btn"
          className="info__close"
          onClick={togglePlayersInfoHidden}
        />
      </div>
      <div className="info__grid">
        <div className="info__section">
          <img src={img} alt="img" className="info__player" />
          <div className="info__overview">
            <span className="info__role">{role}</span>
            <div className="info__wrap">
              <InfoButton img={Target} text={"Crosshair preview"} />
              <div className="info__save" onClick={() => addCrosshair({team, name, crosshair, id})}>
                <img src={RedHeart} alt="copy" />
                <p className="info__text">Save Crosshair settings</p>
              </div>
              <InfoButton img={Computer} text={"Players Resolution"} />
              <div className="info__save">
                <img src={PurpleHeart} alt="copys" />
                <p className="info__text">Save User settings</p>
              </div>
            </div>
          </div>
        </div>

        <Settings settings={settings} />
        <div className="info__footer">
          <Copy crosshair={crosshair}/>
          <p className="info__config">See config</p>
        </div>
      </div>
    </div>
  );
};
const mapStateToProps = createStructuredSelector({
  info: selectPlayersInfo,
  copied: selectIfCopied,
});

const mapDispatchToProps = (dispatch) => ({
  togglePlayersInfoHidden: () => dispatch(togglePlayersInfoHidden()),
  addCrosshair: (item) => dispatch(addCrosshair(item)),
});

export default connect(mapStateToProps, mapDispatchToProps)(PlayerInfo);
