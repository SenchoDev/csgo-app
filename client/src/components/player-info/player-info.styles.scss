.info{
  position: absolute;
  top: 50%;
  left: 50%;                    
  transform: translate(-50%, -50%);
  z-index: 25;
  width: 130rem;
  height: 83rem;
  background-color: #ffffff;
  display: flex;
  flex-direction: column;
  position: fixed;
  
  &__header{
    display: flex;
    padding: 2.2rem 3.2rem;
    align-items: center;
    border-bottom: 1px solid #0A2B47;
  }
  &__heading{
    color: #0A2B47;
    font-size: 2.8rem;
    font-weight: 400;
    margin-right: auto;
  }
  &__close{
    width: 2.4rem;
    height: 2.4rem;
    cursor: pointer;
  }

  &__grid{
    padding: 6rem 11rem;
    display: grid;
    align-self: start;
  }
  &__section{
    display: flex;
    margin-bottom: 4rem;
  
  }
  &__player{
    width: 35rem;
    height: 25rem;
    padding-right: 6rem;
    object-fit: cover;
  }
  &__overview{
    display: flex;
    flex-direction: column;
    padding-top: 4rem;
  }
  &__role{
    background-color: #FD074E;
    font-size: 1.6rem;
    border-radius: 2.5rem;
    width: 6rem;
    height: 3.2rem;
    display: flex;
    justify-content: center;
    align-items: center;
    margin-bottom: 2rem;
  }

  &__wrap{
    display: grid;
    grid-template-rows: 4.5rem 4.5rem;
    grid-template-columns: 27rem 27rem;
    grid-row-gap: 1.7rem;
    grid-column-gap: 2rem;
  }

  &__save{
    display: flex;
    align-items: center;
    cursor:pointer;
  }

  &__text{
    color: #000000;
    font-size: 1.8rem;
    padding-left: 1.5rem;
    &:hover{
      color: #FD074E;
    }
  }
  &__footer{
    padding-top:4rem;
    display: flex;
    align-items: flex-start;
  }
  &__show{
    margin-left: 2rem;
    border-radius: 0.5rem;
    width: 36rem;
    height: 17rem;
    background-color: #FD074E;
    display: none;
    box-shadow: 0 1.5rem 4rem rgba(#000000, 0.15);

    &--text{
      padding: 1rem 2rem;
      font-size: 1.5rem;
    }
    &--toggle{
      display: block;
    }
  }
  &__config{
    margin-top: 1.7rem;
    padding-left: 2rem;
    color: #FD074E;
    font-size: 1.8rem;
    text-decoration: underline;
    cursor: pointer;
  }
  &__copied{
    color: black;
    position: absolute;
    bottom: 2rem;
    right: 2rem;
    display: none;
    width: 25rem;
    height: 6rem;
    background-color: #21AC54;
    font-size: 2.2rem;
    display: none;
    align-items: center;
    justify-content: center;
    border-radius: 2.5rem;
    &-text{
      color: #ffffff;
    }
    &-succes{
      display: flex;
    }
  }

}
