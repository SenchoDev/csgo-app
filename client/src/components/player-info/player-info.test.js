import React from "react";
import { shallow } from "enzyme";

import { PlayerInfo } from './player-info.component'

describe("PlayerInfo component", () => {
  let wrapper;
  let mockTogglePlayersInfoHidden;
  let mockAddCrosshair;
  const mockInfo = { id: 1,
    name: "text",
    img: "jpg",
    role: "text",
    crosshairImg: "jpg",
    settings: {
      monitor: "a",
      mouse: "b",
      keyboard: "c",
      headset: "d",
      mousepad: "e",
    },
    crosshair:"text",}

  beforeEach(() => {
    mockTogglePlayersInfoHidden = jest.fn();

    const mockProps = {
      currentUser: {
        uid: "123",
      },
      togglePlayersInfoHidden: mockTogglePlayersInfoHidden,
    };

    wrapper = shallow(<PlayerInfo {...mockProps} />);
  });

  it('handle toggle players info to null', () => {
    wrapper.find('.info__close').simulate('click');
    
  })

});
