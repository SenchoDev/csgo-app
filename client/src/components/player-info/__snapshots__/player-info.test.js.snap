// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `undefined`;

exports[`PlayerInfo component should render Header component 1`] = `undefined`;
